### PR TITLE
Partition BQ Tables

### PR DESCRIPTION
**Pull request by mtagle**
Partition tables.
Partition rows in these tables by their insertion times.

### CODE REVIEW

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java [hunk 1]
```diff
@@ -83,6 +87,38 @@
     return matches;
   }
 
+  // package private for testing
+  static TableId getPartitionedTableName(TableId baseTableId, Date date) {
+    StringBuilder sb = new StringBuilder();
+    String baseTableName = baseTableId.table();
+    sb.append(baseTableName);
+    sb.append("$");
+    Calendar cal = Calendar.getInstance();
```
Comment (criccomini) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java]: I'm nervous that this code is going to use timezones, and we want everything to be UTC. Can you force it to use UTC?

Comment (criccomini) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java]: Should also move away from `java.util.date` to new Java 8 `java.time` classes.

**Comment by criccomini** [comment]
Also, can you confirm that we don't need to use (and don't get any advantage) from [template tables](https://cloud.google.com/bigquery/streaming-data-into-bigquery#template-tables)? My reading is that we don't need to use that feature, but want to make sure.

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java (hunk 1 above)
Comment (mtagle) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java]: fixed

**Comment by mtagle** [comment]
As a test I just tried inserting into a partitioned table with a template suffix ($20161014) and it didn't work, because table names can't contain special characters like $.  That section doesn't mention anything about partitioned tables, so I'm going to assume that this isn't meant to be used with partitioned tables.

**Comment by criccomini** [comment]
LGTM go ahead and merge in after you've tested locally against a real GCP project. Also, the integration tests should validate as well.

**Pull request merged**